phpunit/functional/Glpi/Form/QuestionType/QuestionTypeUrgencyTest.php

```diff
@@ -61,4 +62,52 @@ public function testUrgencyAnswerIsDisplayedInTicketDescription(): void
             strip_tags($ticket->fields['content']),
         );
     }
+
+    public function testGetUrgencyLevels(): void
+    {
+        /** @var array $CFG_GLPI */
+        global $CFG_GLPI;
+
+        $questionType = new QuestionTypeUrgency();
+        $urgency_levels = array_combine(
+            range(1, 5),
+            array_map(
+                fn($level) => CommonITILObject::getUrgencyName($level),
+                range(1, 5),
+            )
+        );
+
+        // Allow all urgency levels
+        $CFG_GLPI['urgency_mask'] = array_reduce(range(1, 5), fn($mask, $level) => $mask | (1 << $level), 0);
+        $this->assertEquals($urgency_levels, $this->callPrivateMethod($questionType, 'getUrgencyLevels'));
+
+        // Allow only the third urgency level (the third level can't be disabled)
+        $CFG_GLPI['urgency_mask'] = 1 << 3;
+        $this->assertEquals(
+            [3 => $urgency_levels[3]],
+            $this->callPrivateMethod($questionType, 'getUrgencyLevels')
+        );
+
+        // Allow the three first urgency levels
+        $CFG_GLPI['urgency_mask'] = (1 << 1) | (1 << 2) | (1 << 3);
+        $this->assertEquals(
+            array_filter(
+                $urgency_levels,
+                fn($level) => $level <= 3,
+                ARRAY_FILTER_USE_KEY
+            ),
+            $this->callPrivateMethod($questionType, 'getUrgencyLevels'),
+        );
+
+        // Allow the two last urgency levels
+        $CFG_GLPI['urgency_mask'] = (1 << 3) | (1 << 4) | (1 << 5);
+        $this->assertEquals(
+            array_filter(
+                $urgency_levels,
+                fn($level) => $level >= 3,
+                ARRAY_FILTER_USE_KEY
+            ),
```

Comment:
It is best to avoid computation for expected values in a test.
If there are only 3 values to be expected, it is safer to write them down directly rather than filtering an array with untested code.

Non blocking, just a general advice.